management/src/main/java/org/kaazing/gateway/management/monitoring/configuration/impl/MMFMonitoringDataManager.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2007-2014 Kaazing Corporation. All rights reserved.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.kaazing.gateway.management.monitoring.configuration.impl;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.HashSet;
+
+import org.kaazing.gateway.management.monitoring.configuration.MonitorFileWriter;
+import org.kaazing.gateway.management.monitoring.configuration.MonitoringDataManager;
+import org.kaazing.gateway.management.monitoring.entity.manager.impl.ServiceCounterManagerImpl;
+import org.kaazing.gateway.management.monitoring.service.MonitoredService;
+import org.kaazing.gateway.service.MonitoringEntityFactory;
+
+import uk.co.real_logic.agrona.IoUtil;
+
+/**
+ * Implementation of the monitoring MMF manager.
+ */
+public class MMFMonitoringDataManager implements MonitoringDataManager {
+
+    private static final String LINUX = "Linux";
+    private static final String OS_NAME_SYSTEM_PROPERTY = "os.name";
+    private static final String LINUX_DEV_SHM_DIRECTORY = "/dev/shm";
+    private static final String MONITOR_DIR_NAME = "/kaazing";
+
+    private MonitorFileWriter monitorFileWriter;
+    private File monitoringDir;
+    private Collection<MonitoredService> services = new HashSet<>();
```

Comment:
This member variable is only being used to track how many services have been added. So it would be more efficient just to replace it with int serviceCount, starting at 0 and incremented each time addService is called.